raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -88,12 +93,30 @@ def handle_block(
     )
 
     if lock_has_expired:
+        if initiator_state.received_secret_request:
+            reason = 'bad secret request message from target'
```

Comment:
Handling this here means that a payment that failed because of an invalid secret request will take one block longer till it's cancelled!?

Reply:
> Handling this here means that a payment that failed because of an invalid secret request will take one block longer till it's cancelled!?

It may be way more blocks, up to lock timeout, as the off-chain secret request can be sent at any point in time.

This change however makes the failure event consistent with the update in balance, since the byzantine node is the target, which is not necessarily the next hop, and we need to wait for the lock to expire before removing it.

Reply:
You're point makes sense, but on the other hand it might be nice to be able to inform the user as early as possible that the payment failed, even if the funds are still locked.

Reply:
> That sounds like a plan!

So can one of you make an issue for this (so that we don't forget the other PR)?
If I understand correctly you mean to add an event that the payment will fail [here](https://github.com/raiden-network/raiden/pull/3024/files#diff-05f8d80595312a45872095ef4e7a1680L281) ?

Reply:
Added #3044